### DISCUSSION TITLE
MessageBus trigger: Could not publish to topic 'app1-rule-events' for pipeline 'default-pipeline': nats: no response from stream

### DISCUSSION THREAD

**Discussion by edgego** [edgexfoundry/community#169] [General]
When set edgex messagebus to nats-jetstream, and set pipeline to app service configurable, then got error:

level=ERROR ts=2023-08-21T09:16:47.352212598Z app=app- source=messaging.go:239 msg="MessageBus trigger: Could not publish to topic 'app1-rule-events' for pipeline 'default-pipeline': nats: no response from stream"
level=ERROR ts=2023-08-21T09:16:47.352313369Z app=app- source=triggermessageprocessor.go:191 msg="failed to process output for message '27b943f2-9978-441c-b726-6ac651694e5e' on pipeline default-pipeline: nats: no response from stream"
level=ERROR ts=2023-08-21T09:16:47.352364611Z app=app- source=messaging.go:199 msg="MessageBus Trigger: Failed to process message on pipeline(s): 1 error occurred:\n\t* nats: no response from stream\n\n"
level=ERROR ts=2023-08-21T09:16:48.370327233Z app=app- source=messaging.go:239 msg="MessageBus trigger: Could not publish to topic 'app1-rule-events' for pipeline 'default-pipeline': nats: no response from stream"

app1 messagebus config:

[Writable]
  LogLevel = "INFO"
  [Writable.StoreAndForward]
    Enabled = true
    RetryInterval = "5m"
    MaxRetryCount = 10
  [Writable.Pipeline]
    UseTargetTypeOfByteArray = false
    ExecutionOrder = "FilterByDeviceName,SetResponseData"
    [Writable.Pipeline.Functions.FilterByDeviceName]
      [Writable.Pipeline.Functions.FilterByDeviceName.Parameters]
        DeviceNames = "mod1"
        FilterOut ="false"
    [Writable.Pipeline.Functions.SetResponseData]
      [Writable.Pipeline.Functions.SetResponseData.Parameters]
        ResponseContentType="application/json"
[Writable.InsecureSecrets]
  [Writable.InsecureSecrets.DB]
      path = "redisdb"
      [Writable.InsecureSecrets.DB.Secrets]
        username = ""
        password = ""

[Trigger]
Type="edgex-messagebus"
[Trigger.EdgexMessageBus]
Type = "nats-jetstream"
[Trigger.EdgexMessageBus.SubscribeHost]
Host = "edge-nats-server"
Port = 4222
Protocol = "tcp"
SubscribeTopics="edgex/events/#"
[Trigger.EdgexMessageBus.PublishHost]
Host = "edge-nats-server"
Port = 4222
Protocol = "tcp"
PublishTopic="app1-rule-events"
[Trigger.EdgexMessageBus.Optional]
authmode = 'usernamepassword'
secretname = 'redisdb'
ClientId ="app1"
Format = "nats"
ConnectTimeout = "5"
RetryOnFailedConnect = "true"
QueueGroup = ""
Durable = ""
AutoProvision = "true"
Deliver = "new"
DefaultPubRetryAttempts = "2"
Subject = "edgex/#"

**Comment by lenny-goodell** [comment]
@edgego , I assume you are using Levski release?

Have you read the NATs MessageBus section here: https://docs.edgexfoundry.org/2.3/microservices/general/messagebus/#nats
Specifically the part about Currently services must be specially built with the include_nats_messaging tag to enable this option ? This applies to all services, not just your App Service and The NATS Server must be running. See deployment section
https://docs.edgexfoundry.org/2.3/microservices/general/messagebus/#docker_1

**Reply by edgego** [reply]
@lenny-intel  Yes, i am using Levski  release. And built docker image with  include_nats_messaging. And NATS Server  is running.

**Reply by lenny-goodell** [reply]
@edgego , did you build ALL edgex services with NATS?
How are you running the services?

**Reply by edgego** [reply]
Dear @lenny-intel   after restarting , sound like work fine. Thx!
